transport/http/encode_decode.go

```diff
@@ -17,6 +17,12 @@ type DecodeRequestFunc func(context.Context, *http.Request) (request interface{}
 // encodes the object directly to the request body.
 type EncodeRequestFunc func(context.Context, *http.Request, interface{}) error
 
+// CreateRequestFunc creates an outgoing HTTP request based on the passed
+// request object. It's designed to be used in HTTP clients, for client-side
+// endpoints. It's a more powerful version of EncodeRequestFunc, and can be used
+// if more fine-grained control of the HTTP request is required.
+type CreateRequestFunc func(context.Context, interface{}) (*http.Request, error)
```

Comment:
Don't you want to mark `EncodeRequestFunc` as `// deprecated` linking [this issue](https://github.com/go-kit/kit/issues/796) for more information ?

Reply:
No, because I don't consider it deprecated, this is just an alternative.